assets/js/admin-settings/index.jsx

```diff
@@ -0,0 +1,118 @@
+/* global localizedData */
+import {
+  Button,
+  SelectControl,
+  TextControl,
+  ToggleControl,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import React, { useEffect, useState } from 'react';
+import useSiteOptions from '../services/hooks/use-site-options';
+import Rule from './rule';
+import { ruleCorral } from './styles';
+
+
+const AdminSettings = () => {
+  const [{ loading, saving, settings }, setOptions] = useSiteOptions();
+  const [originIndex, setOriginIndex] = useState(null);
+  const [targetIndex, setTargetIndex] = useState(null);
+
+  /**
+   * Helper function for saving the in-memory settings to the server.
+   */
+   const saveSettings = (updatedRules) => {
+    const next = { ...settings, apple_news_automation: updatedRules };
+
+    // Enforce some defaults prior to save.
+    // Request will 500 when site_logo === null.
+    next.site_logo = next.site_logo ?? 0;
+    console.log('settings-pre-save', next);
+
+    // Kick off the save to the server.
+    setOptions(next);
+  };
+  
+  const cleanData = () => {
+    saveSettings([]);
+  }
+  
+  // useEffect(() => {
+  //   cleanData();
+  // },[]);
+
+  const addRule = (newRule) => {
+    const updatedRules = settings.apple_news_automation ?? [];
+    updatedRules.push(newRule);
+    saveSettings(updatedRules);
+  }
+
+  const deleteRule = (ruleIndex) => {
+    const oldRules = settings.apple_news_automation ?? [];
+    const updatedRules = oldRules.filter((x, index) => index !== ruleIndex);
+    saveSettings(updatedRules);
+  }
+
+  const reorderRule = () => {
+    // Do nothing if the rule it dropped into its own slot.
+    if (originIndex === targetIndex) {
+      return;
+    }
+    const updatedRules = settings.apple_news_automation ?? [];
```

Comment:
This should make a copy of the array before operating on it, so spread what's returned:
```suggestion
    const updatedRules = [...(settings.apple_news_automation ?? [])];
```